File: saml-attribute-provider-archetype/src/main/resources/archetype-resources/src/main/java/__pluginName__/__pluginName__SamlAttributeProvider.java
package ${package}.${pluginName};

import se.curity.identityserver.sdk.attribute.Attributes;
import se.curity.identityserver.sdk.attribute.AuthenticationAttributes;
import se.curity.identityserver.sdk.saml.SamlAttributeProvider;
import se.curity.identityserver.sdk.saml.RequestedSamlAttribute;
import se.curity.identityserver.sdk.saml.SamlServiceProvider;

import java.util.Set;

public final class ${pluginName}SamlAttributeProvider implements SamlAttributeProvider
{
    private final ${pluginName}SamlAttributeProviderConfig _configuration;

    public ${pluginName}SamlAttributeProvider(${pluginName}SamlAttributeProviderConfig configuration)
    {
        _configuration = configuration;

        /**
         * SAML attribute provider behave similarly to claims providers. You can see
         * examples of claims providers on our GitHub: https://github.com/search?q=topic%3Aclaims-provider+org%3Acurityio
         */
    }

    @Override
    public Attributes getAttributeValues(Set<RequestedSamlAttribute> requestedSamlAttributes,
                                     AuthenticationAttributes userAuthenticationAttributes,
                                     SamlServiceProvider serviveProvider)
    {
        throw new UnsupportedOperationException();
    }
}

File: kotlin-saml-attribute-provider-archetype/src/main/resources/archetype-resources/src/main/kotlin/__pluginName__/__pluginName__SamlAttributeProvider.kt
package ${package}.${pluginName}

import se.curity.identityserver.sdk.attribute.Attributes
import se.curity.identityserver.sdk.attribute.AuthenticationAttributes
import se.curity.identityserver.sdk.saml.SamlAttributeProvider
import se.curity.identityserver.sdk.saml.RequestedSamlAttribute
import se.curity.identityserver.sdk.saml.SamlServiceProvider

class ${pluginName}SamlAttributeProvider(private val _configuration: ${pluginName}SamlAttributeProviderConfig): SamlAttributeProvider
{
    /**
     * SAML attribute provider behave similarly to claims providers. You can see
     * examples of claims providers on our GitHub: https://github.com/search?q=topic%3Aclaims-provider+org%3Acurityio
     */

    override fun getAttributeValues(requestedSamlAttributes: Set<RequestedSamlAttribute>,
                                userAuthenticationAttributes: AuthenticationAttributes,
                                serviceProvider: SamlServiceProvider): Attributes
    {
        TODO()
    }
}
